toolkit/tools/pkg/isomakerlib/isomaker.go

```diff
@@ -540,48 +759,64 @@ func (im *IsoMaker) buildIsoImageFilePath() string {
 
 // deferIsoMakerCleanUp accepts clean-up tasks to be ran when the entire
 // build process has finished, NOT at the end of the current scope.
-func (im *IsoMaker) deferIsoMakerCleanUp(cleanUpTask func()) {
+func (im *IsoMaker) deferIsoMakerCleanUp(cleanUpTask func()(error)) {
 	im.isoMakerCleanUpTasks = append(im.isoMakerCleanUpTasks, cleanUpTask)
 }
 
 // isoMakerCleanUp runs all clean-up tasks scheduled through "deferIsoMakerCleanUp".
 // Tasks are ran in reverse order to how they were scheduled.
-func (im *IsoMaker) isoMakerCleanUp() {
+func (im *IsoMaker) isoMakerCleanUp() (error err) {
 	// We defer again to run the tasks in the correct order AND so that a potential
 	// panic from one of these doesn't prevent other ones from running.
 	for _, cleanUpTask := range im.isoMakerCleanUpTasks {
-		defer cleanUpTask()
+		defer func() {
```

Comment:
The defer here might no longer be neccessary.

Reply:
When tasks are added for later clean-up, they are added to an array (see `deferIsoMakerCleanUp`).
When they are executed, it is desired that they are executed in the opposite order.
The original implementation relied on defer to do that reversal.
We can traverse the array also in the opposite order during this clean-up.
I find traversing the array in the reverse order more readable. But I'm okay either way...
@cwize1 Do you have a preference?

Reply:
Yeah, I’d also prefer being explicit and iterating the list in reverse.

Reply:
+1 for reverse iteration. I think with the previous use of panics a `defer` was just necessary.

Reply:
thx - updated.